.github/workflows/qubes-dom0-packagev2.yml

```diff
@@ -68,10 +75,69 @@ jobs:
           sed -i "s#^  prefix: fepitre/qubes-#  prefix: QubesOS/qubes-#" builder.yml
           sed -i "s#^  branch: builderv2#  branch: main#" builder.yml
           sed -i "s#^artifacts-dir: .*#artifacts-dir: $PWD/artifacts#" builder.yml
-          sed -i "1,/^  - ${{ inputs.qubes-component }}/s#^  - ${{ inputs.qubes-component }}#&:#" builder.yml
-          sed -i "/^  - ${{ inputs.qubes-component }}:/a\      verification-mode: insecure-skip-checking" builder.yml
-          sed -i "/^  - ${{ inputs.qubes-component }}:/a\      branch: ${${{ github.ref_type }}}" builder.yml
-          sed -i "/^  - ${{ inputs.qubes-component }}:/a\      url: ${URL/git:/https:}" builder.yml
+
+          branch_name=${${{ github.ref_type }}}
+          if [ -z "$branch_name" ]; then
+              # github.head_ref is set only for pull requests, this should
+              # handle pushes
+              branch_name=$(basename "$GITHUB_REF")
+          fi
+
+          if [ -n "$PKG_DIR" ]; then
+              #
+              # 1. Clone repository locally
+              # 2. Add files required by qubes-builderv2
+              # 3. Commit them
+              # 4. Put local path as a repository URL to configuration file
+              #
+
+              # `plugins/fetch` is copied inside Docker container, place clone there
+              rel_clone_dir=plugins/fetch/tmp_clone
+              clone_dir=qubesbuilder/$rel_clone_dir
+              git clone --depth 1 --branch "$branch_name" "${URL/git:/https:}" "$clone_dir"
+
+              # qubes-builderv2 expects top directory of extracted sources to
+              # follow "name-version" naming scheme and will fail if other
+              # naming is used
+              #
+              # At the same time when generating a source archive, it uses
+              # naming scheme from `Source0`, which inevitably leads to a
+              # failure if it's not what qubes-builderv2 expects.
+              sed '/^Source0:/s/\t.*/\t%{name}-%{version}.tar.gz/' \
```

Comment:
Why are the following files created here if they are included in https://github.com/TrenchBoot/secure-kernel-loader/pull/14? Which ones are used?

Reply:
I thought that a package that builds locally can be integrated into CI, but that didn't work.  So I've added a working package to SKL repo and made CI reuse its spec file.  `rel` is trivial, `version` needs to be overwritten anyway, `.qubesbuilder` with `source` key wouldn't work for CI as is (due to presence of `source` key which changes behaviour of the builder) and is easier to generate than to edit.